File: Daily Challenge/C++/Ascending Descending 9.8.2025.cpp
#include <iostream>
#include <vector>
using namespace std;

bool is_asc(const vector<int>& v) {
    for (int i = 1; i < v.size(); ++i)
        if (v[i] < v[i-1]) return false;
    return true;
}

bool is_desc(const vector<int>& v) {
    for (int i = 1; i < v.size(); ++i)
        if (v[i] > v[i-1]) return false;
    return true;
}

bool all_equal(const vector<int>& v) {
    for (int i = 1; i < v.size(); ++i)
        if (v[i] != v[0]) return false;
    return true;
}

void print(const vector<int>& v) {
    for (int x : v) cout << x << " ";
    cout << endl;
}

int main() {
    int m, n;
    cin >> m;
    vector<int> a(m);
    for (int i = 0; i < m; ++i) cin >> a[i];

    cin >> n;
    vector<int> b(n);
    for (int i = 0; i < n; ++i) cin >> b[i];

    if (all_equal(a) && all_equal(b)) {
        if (a[0] < b[0]) { print(a); print(b); }
        else if (b[0] < a[0]) { print(b); print(a); }
        else { print(a); print(b); }
    } else if (is_asc(a) && is_desc(b)) {
        print(a); print(b);
    } else if (is_desc(a) && is_asc(b)) {
        print(b); print(a);
    } else {
        print(a); print(b);
    }

    return 0;
}

File: Daily Challenge/Java/Ascending Descending 9.8.2025.java
import java.util.*;

public class Main {
    static boolean isAsc(List<Integer> list) {
        for (int i = 1; i < list.size(); i++)
            if (list.get(i) < list.get(i - 1)) return false;
        return true;
    }

    static boolean isDesc(List<Integer> list) {
        for (int i = 1; i < list.size(); i++)
            if (list.get(i) > list.get(i - 1)) return false;
        return true;
    }

    static boolean allEqual(List<Integer> list) {
        for (int i = 1; i < list.size(); i++)
            if (!list.get(i).equals(list.get(0))) return false;
        return true;
    }

    static void print(List<Integer> list) {
        for (int x : list) System.out.print(x + " ");
        System.out.println();
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int m = sc.nextInt();
        List<Integer> a = new ArrayList<>();
        for (int i = 0; i < m; i++) a.add(sc.nextInt());

        int n = sc.nextInt();
        List<Integer> b = new ArrayList<>();
        for (int i = 0; i < n; i++) b.add(sc.nextInt());

        if (allEqual(a) && allEqual(b)) {
            if (a.get(0) < b.get(0)) { print(a); print(b); }
            else if (b.get(0) < a.get(0)) { print(b); print(a); }
            else { print(a); print(b); }
        } else if (isAsc(a) && isDesc(b)) {
            print(a); print(b);
        } else if (isDesc(a) && isAsc(b)) {
            print(b); print(a);
        } else {
            print(a); print(b);
        }
    }
}
